### PR TITLE
add TextTemplateService

### PR DESCRIPTION
**Pull request by grinakate**
Add service to TextTemplateService github repository
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/947

- add solution with name "LT.DigitalOffice.TextTemplateService";
- add directory with name "Solution items" and put the following files there: ".dockerignore", ".editorconfig", ".gitignore", "Common.Configurations.targets", "Dockerfile", "main.yml", "README.md"
- add directory with name "src" and put the following items:
application with name "TextTemplateService" (add appsettings.json, set Program.cs, Startup.cs), 
add following DLLs: TextTemplateService.Broker (add "Consumers" directory), 
TextTemplateService.Business (add "Commands" directory), 
TextTemplateService.Data (add "Interfaces" directory), 
TextTemplateService.Data.Provider (add "IDataProvider" interface, 
TextTemplateService.Data.Provider.MsSql.Ef (add "Migrations" directory and "TextTemplateServiceDbContext" class), 
TextTemplateService.Mappers, 
TextTemplateService.Models.Db, 
TextTemplateService.Models.Dto, 
TextTemplateService.Validation

### CODE REVIEW

File: Common.Configurations.targets [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<Project>
+	<PropertyGroup>
+		<TargetFramework>net6.0</TargetFramework>
+
```
Comment (ndoriomedov) [inline Common.Configurations.targets]: remove empty line

File: src/TextTemplateService/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        GetString("RabbitMqPassword", _rabbitMqConfig.Password, _serviceInfoConfig.Id, "Password"));
+    }
+
+    private void ConfigureMassTransit(IServiceCollection services)
+    {
+      (string username, string password) = GetRabbitMqCredentials();
+
+      services.AddMassTransit(x =>
+      {
+
```
Comment (ndoriomedov) [inline src/TextTemplateService/Startup.cs]: empty line

File: src/TextTemplateService/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        x.UsingRabbitMq((context, cfg) =>
+        {
+          cfg.Host(_rabbitMqConfig.Host, "/", host =>
+          {
+            host.Username(username);
+            host.Password(password);
+          });
+
```
Comment (ndoriomedov) [inline src/TextTemplateService/Startup.cs]: empty line)

File: src/TextTemplateService/TextTemplateService.csproj [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
+
+  <ItemGroup>
+    <PackageReference Include="AspNetCore.HealthChecks.UI.Client" Version="5.0.1" />
+    <PackageReference Include="AspNetCore.HealthChecks.SqlServer" Version="5.0.2" />
```
Comment (ndoriomedov) [inline src/TextTemplateService/TextTemplateService.csproj]: update to latest rc version (in manage nugets window set "Include prerelease check" )

File: src/TextTemplateService/appsettings.Production.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+﻿{
+  "Serilog": {
+  },
+  "ConnectionStrings": {
+    "SQLConnectionString": "Server=dbTextTemplate-sql-server;User=sa;Password=TextTemplate_1234;Database=TextTemplateDB;"
+  },
+  "RabbitMQ": {
+    "Host": "lt_do_rabbitmq"
+  },
```
Comment (ndoriomedov) [inline src/TextTemplateService/appsettings.Production.json:9]: add MemoryCache lifetime maybe?

File: src/TextTemplateService/appsettings.json [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    "GetSmtpCredentialsEndpoint": "CompanyService_GetSmtpCredentials",
+
+    // OfficeService
+    "GetOfficesEndpoint": "OfficeService_GetOffices",
+
+    // EmailService
+    "CreateSmtpCredentialsEndpoint": "EmailService_CreateSmtpCredentials",
+
+    // ImageService
+    "GetImagesEndpoint": "ImageService_GetImages"
```
Comment (ndoriomedov) [inline src/TextTemplateService/appsettings.json]: do we need all these endpoints?

File: src/TextTemplateService.Data.Provider.MsSql.Ef/TextTemplateServiceDbContext.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+namespace LT.DigitalOffice.TextTemplateService.Data.Provider.MsSql.Ef
+{
+  public class TextTemplateServiceDbContext : DbContext, IDataProvider
+  {
+    public TextTemplateServiceDbContext(DbContextOptions<TextTemplateServiceDbContext> options)
+    : base(options)
+    {
+    }
+
+    // Fluent API is written here.
```
Comment (Cheephot) [inline src/TextTemplateService.Data.Provider.MsSql.Ef/TextTemplateServiceDbContext.cs]: remove

File: src/TextTemplateService/Program.cs [hunk 8]
```diff
@@ -0,0 +1,59 @@
+using Microsoft.AspNetCore.Hosting;
```
Comment (Cheephot) [inline src/TextTemplateService/Program.cs:1]: fixe all tabs in this cs file

File: src/TextTemplateService/Startup.cs [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+using StackExchange.Redis;
+using System;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace LT.DigitalOffice.TextTemplateService
+{
+  public class Startup : BaseApiInfo
+  {
+
```
Comment (Cheephot) [inline src/TextTemplateService/Startup.cs]: remove line

File: src/TextTemplateService/Startup.cs [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+      if (string.IsNullOrEmpty(connStr))
+      {
+        connStr = Configuration.GetConnectionString("SQLConnectionString");
+        Log.Information($"SQL connection string from appsettings.json was used. Value '{HidePasswordHelper.HidePassword(connStr)}'.");
+      }
+      else
+      {
+        Log.Information($"SQL connection string from environment was used. Value '{HidePasswordHelper.HidePassword(connStr)}'.");
+      }
+
```
Comment (Cheephot) [inline src/TextTemplateService/Startup.cs:107]: remove line

File: src/TextTemplateService/Startup.cs [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+            { HealthStatus.Unhealthy, 200 },
+            { HealthStatus.Healthy, 200 },
+            { HealthStatus.Degraded, 200 },
+          },
+          Predicate = check => check.Name != "masstransit-bus",
+          ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse
+        });
+      });
+    }
+
```
Comment (Cheephot) [inline src/TextTemplateService/Startup.cs:171]: remove line

File: Common.Configurations.targets (hunk 1 above)
Comment (grinakate) [reply Common.Configurations.targets]: done

File: src/TextTemplateService/Startup.cs (hunk 2 above)
Comment (grinakate) [reply src/TextTemplateService/Startup.cs]: fixed

File: src/TextTemplateService/Startup.cs (hunk 3 above)
Comment (grinakate) [reply src/TextTemplateService/Startup.cs]: fixed

File: src/TextTemplateService/TextTemplateService.csproj (hunk 4 above)
Comment (grinakate) [reply src/TextTemplateService/TextTemplateService.csproj]: done

File: src/TextTemplateService/appsettings.Production.json (hunk 5 above)
Comment (grinakate) [reply src/TextTemplateService/appsettings.Production.json:9]: done

File: src/TextTemplateService.Data.Provider.MsSql.Ef/TextTemplateServiceDbContext.cs (hunk 7 above)
Comment (grinakate) [reply src/TextTemplateService.Data.Provider.MsSql.Ef/TextTemplateServiceDbContext.cs]: done

File: src/TextTemplateService/Program.cs (hunk 8 above)
Comment (grinakate) [reply src/TextTemplateService/Program.cs:1]: done

File: src/TextTemplateService/Startup.cs (hunk 9 above)
Comment (grinakate) [reply src/TextTemplateService/Startup.cs]: done

File: src/TextTemplateService/Startup.cs (hunk 10 above)
Comment (grinakate) [reply src/TextTemplateService/Startup.cs:107]: done

File: src/TextTemplateService/Startup.cs (hunk 11 above)
Comment (grinakate) [reply src/TextTemplateService/Startup.cs:171]: done

**Review by ekaterinaryzhenkova — APPROVED** [review]

File: Common.Configurations.targets [hunk 12]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<Project>
+	<PropertyGroup>
```
Comment (VladSpitsin) [inline Common.Configurations.targets]: fix tabs in this file

File: src/TextTemplateService.Business/TextTemplateService.Business.csproj [hunk 13]
```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
```
Comment (VladSpitsin) [inline src/TextTemplateService.Business/TextTemplateService.Business.csproj:2]: fix tabs, same in other .csproj files

**Review by VladSpitsin — APPROVED** [review]

**Review by irlirla — APPROVED** [review]

File: Common.Configurations.targets [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+	<PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
+		<DebugSymbols>false</DebugSymbols>
+		<DebugType>none</DebugType>
+		<Optimize>true</Optimize>
+		<DefineConstants>RELEASE;TRACE</DefineConstants>
+	</PropertyGroup>
+
+	<ItemGroup Condition="$(AssemblyName.Contains('.UnitTests')) != 'true'">
+		<PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.51" />
```
Comment (MariyaSin) [inline Common.Configurations.targets]: update to 53version and add kernel.BrockerSupport

File: src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+using LT.DigitalOffice.Models.Broker.Requests.Office;
+using LT.DigitalOffice.Models.Broker.Requests.Position;
+using LT.DigitalOffice.Models.Broker.Requests.Project;
+using LT.DigitalOffice.Models.Broker.Requests.User;
+
+namespace LT.DigitalOffice.TextTemplateService.Models.Dto.Configuration
+{
+  public class RabbitMqConfig : BaseRabbitMqConfig
+  {
+    // department
```
Comment (MariyaSin) [inline src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs]: you can remove department endpoints

File: src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+
+    [AutoInjectRequest(typeof(IGetDepartmentsRequest))]
+    public string GetDepartmentsEndpoint { get; set; }
+
+    // position
```
Comment (MariyaSin) [inline src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs]: remove positions endpoints

File: src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+
+    [AutoInjectRequest(typeof(IGetPositionsRequest))]
+    public string GetPositionsEndpoint { get; set; }
+
+    // office
```
Comment (MariyaSin) [inline src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs]: remove offices endpoints

File: src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+
+    [AutoInjectRequest(typeof(IGetOfficesRequest))]
+    public string GetOfficesEndpoint { get; set; }
+
+    // user
```
Comment (MariyaSin) [inline src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs]: remove users endpoints

File: src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+
+    [AutoInjectRequest(typeof(IGetUsersDataRequest))]
+    public string GetUsersDataEndpoint { get; set; }
+
+    [AutoInjectRequest(typeof(ICheckUsersExistence))]
+    public string CheckUsersExistenceEndpoint { get; set; }
+
+    // image
```
Comment (MariyaSin) [inline src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs]: remove images endpoints

File: src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+
+    [AutoInjectRequest(typeof(IGetImagesRequest))]
+    public string GetImagesEndpoint { get; set; }
+
+    // project
```
Comment (MariyaSin) [inline src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs]: remove projectsEndpoints

File: src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+
+    [AutoInjectRequest(typeof(IGetProjectsRequest))]
+    public string GetProjectsEndpoint { get; set; }
+
+    // email
+
+    [AutoInjectRequest(typeof(IGetSmtpCredentialsRequest))]
```
Comment (MariyaSin) [inline src/TextTemplateService.Models.Dto/Configuration/RabbitMqConfig.cs]: remove the endpoint

File: src/TextTemplateService/appsettings.json [hunk 22]
```diff
@@ -0,0 +1,55 @@
...
+  "ConnectionStrings": {
+    "Redis": "localhost:6379"
+  },
+  "RabbitMQ": {
+    //Common
+    "DisactivateUserEndpoint": "Common_DisactivateUser",
+
+    //UserService
+    "CheckUserIsAdminEndpoint": "UserService_GetUserAdminStatus",
+    "GetUsersDataEndpoint": "UserService_GetUsersData",
```
Comment (MariyaSin) [inline src/TextTemplateService/appsettings.json]: remove

File: src/TextTemplateService/appsettings.json [hunk 23]
```diff
@@ -0,0 +1,55 @@
...
+    "CheckUsersExistenceEndpoint": "UserService_CheckUsersExistence",
```
Comment (MariyaSin) [inline src/TextTemplateService/appsettings.json]: remove

File: src/TextTemplateService/appsettings.json [hunk 24]
```diff
@@ -0,0 +1,55 @@
...
+
+    //RigthService
+    "CheckUserRightsEndpoint": "CRService_CheckUserRights",
+
+    //AuthService
+    "ValidateTokenEndpoint": "AuthService_ValidateToken",
+
+    //ImageService
+    "GetImagesEndpoint": "ImageService_GetImages"
```
Comment (MariyaSin) [inline src/TextTemplateService/appsettings.json]: remove

File: Common.Configurations.targets [hunk 25]
```diff
@@ -0,0 +1,47 @@
...
+
+	<PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
+		<DebugSymbols>false</DebugSymbols>
+		<DebugType>none</DebugType>
+		<Optimize>true</Optimize>
+		<DefineConstants>RELEASE;TRACE</DefineConstants>
+	</PropertyGroup>
+
+	<ItemGroup Condition="$(AssemblyName.Contains('.UnitTests')) != 'true'">
+        <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.53" />
```
Comment (MariyaSin) [inline Common.Configurations.targets]: fix tabs

File: src/TextTemplateService/appsettings.json [hunk 26]
```diff
@@ -0,0 +1,50 @@
...
+  "ServiceInfo": {
+    "Name": "TextTemplateService",
+    "Id": "E50F762C-F8CF-41E8-B81C-F5CFEF8D68C8"
+  },
+  "ConnectionStrings": {
+    "Redis": "localhost:6379"
+  },
+  "RabbitMQ": {
+    //Common
+    "DisactivateUserEndpoint": "Common_DisactivateUser",
```
Comment (MariyaSin) [inline src/TextTemplateService/appsettings.json]: you can  remove it

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**